src/pages/api/event/[id]/repair-request.ts

```diff
@@ -26,16 +43,69 @@ async function getRepairRequests(
     throw new ApiError(HttpStatusCode.NotFound, "Event not found");
   }
 
-  const repairRequests = await prisma.repairRequest.findMany({
-    where: { event: { id: eventId } },
-    include: {
-      images: true
-    }
-  });
+  const sortObj: Record<string, SortDirection> = {
+    [sortKey]: sortMethod
+  };
+
+  // Find user ids that match the search
+  // TODO: if there are more than 500 users that match the search, results will be cut off
+  let userIdList = undefined;
+  if (searchWord) {
+    const users = await userService.getMany({
+      query: searchWord,
+      perPage: 500,
+      orderBy: "created_at",
+      page: 1
+    });
+
+    userIdList = users.items.map((u) => u.id);
+  }
+
+  // Query DB
+  const where: Prisma.RepairRequestWhereInput = {
+    event: { id: eventId as string },
```

Comment:
Could remove this cast.

Reply:
The cast exists to catch any errors in the where and provide autocomplete since the type is not inferred as its outside of a prisma query